main.js

```diff
@@ -90,8 +90,24 @@ window.dataLayer = window.dataLayer || [];
     return "" + Math.floor(time/60/60) + ":" + (Math.floor((time/60))%60 < 10 ? "0" : "") + Math.floor((time/60))%60;
   }
 
-  const calculateGoal = function() {
-    const date = new Date();
+  const reliableDate = async function() {
+    try {
+      return await dateEdt()
+    } catch {
+      console.log("catch activated, reverting to system time!")
+      return new Date()
+    }
+  }

```

Comment:
For testing based on local time, I think this code would be perfect for your use! This statement switches between local and global time based on network availability, so with some modification it could be toggled manually.